R/zchunk_L100.Population_downscale_ctry.R

```diff
@@ -33,54 +34,191 @@ module_socioeconomics_L100.Population_downscale_ctry_DISABLED <- function(comman
+      filter(year != socioeconomics.AGG_BASE_YEAR) %>% # Remove base aggregation year (it's in the next step, don't want to duplicate it)
+      select(Maddison_ctry, iso, year, pop_scale)
+    # Bind these with the other countries
+    reg_scaled <- as_tibble(pop_thous_ctry_reg) %>%
+      left_join(agg_ratio, by = c("Maddison_ctry", "iso", "year")) %>%
+      mutate(pop = if_else((year < socioeconomics.AGG_BASE_YEAR & !is.na(pop_scale)), pop_scale, pop)) %>% # Replace NA values for countries with downscaled population values
+      mutate(iso = if_else(Maddison_ctry == "World Total", "world_total", iso)) %>% # Will need total world population in step 4
+      filter(!is.na(iso)) %>%  # Remove rows with no iso code (generally aggregate regions)
+      select(iso, year, pop)  # Keep only necessary variables
+
+    # Third, interpolate available population data to generate values for required historical years (1700, 1750, 1800, 1850, 1900)
+    hist_interp <- tibble(year = socioeconomics.MADDISON_HISTORICAL_YEARS) %>%  # Create tibble with required years (not all are in the original Maddison data)
+      full_join(reg_scaled, by = "year") %>%  # Join with available population data (creates NAs for years not available in raw data)
+      complete(year, nesting(iso)) %>%  # Completes tibble to include all years for all iso (creates missing values)
+      filter(!is.na(iso)) %>%  # iso NA values created with complete on year and iso table
+      arrange(iso, year) %>%  # For easier readability when checking results
+      group_by(iso) %>%
+      mutate(pop = approx_fun(year, pop)) %>%  # Interpolate -- note that there will still be missing values for countries that do not have end values (1500, 1600, or 1700)
+      mutate(pop = if_else(is.na(pop) & year == 1800, pop[year == 1820], pop)) %>%  # Replace missing 1800 values with 1820 for countries that begin in 1820 (note that Panama has a value in 1820 = 0, not NA)
+      filter(year %in% c(socioeconomics.MADDISON_HISTORICAL_YEARS, socioeconomics.AGG_BASE_YEAR)) # Keep only needed years
+
+    # Fourth, assign population values to countries with missing values in historical years
+    # Extract total world population for historic years
+    pop_global <- filter(hist_interp, iso == "world_total") %>%
+      ungroup() %>%
+      select(-iso)
+    # Sum all countries' populations in each period
+    pop_notmissing <- filter(hist_interp, iso != "world_total") %>%
+      group_by(year) %>%
+      mutate(pop = if_else(is.na(pop), 0, pop)) %>%
+      summarize(pop = sum(pop))
+    # Subtract reported countries from total global - these are the total population values in each period that will be allocated to countries with missing values
+    pop_missing <- left_join(pop_global, pop_notmissing, by = "year") %>%
+      mutate(pop_allocate = pop.x - pop.y) %>%
+      select(year, pop_allocate) %>%
+      mutate(year = paste("X", year, sep = "")) %>%
```

Comment:
I started to flip out but, looking below, think there's a reasonable argument for going back to wide and using the X notation for easy referencing. But would be useful to have a comment explaining _why_ you're doing this.